tests/chainer_tests/test_sequential.py

```diff
@@ -607,4 +581,148 @@ def test_empty_sequential(self):
             seq(x)
 
 
+class TestSequentialSerialization(unittest.TestCase):
+
+    # MyLink is a link that depends on a parameter whose value is filled
+    # with an integer specified by `n`.
+
+    # `fn_create` is a test-specific function to create a target link.
+    # `fn_create` has one argument `nth`, which is either 0 or 1, each of
+    # which represents before or after serialization.
+    # `n` argument of `MyLink` is dependent on this `nth`, thus the link is
+    # initialized with different values depending on before/after
+    # serialization.
+
+    # If serialization did not work correctly, the link would have a different
+    # value after serialization, causing tests to fail.
+
+    class MyLink(chainer.Link):
+
+        def __init__(self, n):
+            chainer.Link.__init__(self)
+            with self.init_scope():
+                self.n = chainer.Parameter(
+                    numpy.full((1, 2), n, dtype=numpy.float32))
+
+        def forward(self, x):
+            return chainer.functions.vstack((x, self.n))
+
+    class MyFunc(chainer.FunctionNode):
+        def __init__(self, n):
+            chainer.FunctionNode.__init__(self)
+            self.n = n
+
+        def forward(self, inputs):
+            x, = inputs
+            y = numpy.vstack((
+                x, numpy.full((1, 2,), self.n, dtype=x.dtype)))
+            return y,
+
+        backward = None  # not used in this test
+
+    def my_func(self, n):
+        # defined as function-local function
+        def func_impl(x):
+            y, = self.MyFunc(n).apply((x,))
+            return y
+        return func_impl
+
+    def serialize(self, seq):
+        f = six.BytesIO()
+        chainer.serializers.save_npz(f, seq)
+        return f.getvalue()
+
+    def deserialize(self, seq, bytes):
+        f = six.BytesIO(bytes)
+        chainer.serializers.load_npz(f, seq)
+
+    def check_serialization(self, fn_create):
+        x = numpy.array([[100, 100]], numpy.float32)
+        seq1 = fn_create(0)
+
+        y1 = seq1(x)
+        if len(seq1) == 0:
```

Comment:
Calling an empty `Sequential` is forbidden since #6304.

Reply:
Thank you for pointing out. Fixed.